### PR TITLE
Refactor | Create proposal with Yup and Formik

### LINKED ISSUE #1018: Refactor: Proposal Creation ETH resolve address bug

### PR DESCRIPTION
**Pull request by Da-Colon**
## Description
This PR covers the refactor of the create proposal form to utilize Yup and Formik for validation and form control. No more infinite renders!

## Notes
- Made some typing name changes
- removed all the other form controls

## Issue / Notion doc (if applicable)
Closes #1018 

## Testing
Successfully create Multsig/Usul Proposals:
- Look for nonce changes are still picked up
- metadata is still executed properly

### CODE REVIEW

File: src/utils/contract.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { utils } from 'ethers';
+
+export const encodeFunction = (
```
Comment (herbig) [inline src/utils/contract.ts:3]: we could probably put this in` crypto.ts` instead of a new file

File: src/utils/contract.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  let functionSignature = `function ${_functionName}`;
+  if (_functionSignature) {
+    functionSignature = functionSignature.concat(`(${_functionSignature})`);
+  } else {
+    functionSignature = functionSignature.concat('()');
+  }
+  const parameters = !!_parameters ? _parameters.split(',').map(p => (p = p.trim())) : undefined;
+  try {
+    return new utils.Interface([functionSignature]).encodeFunctionData(_functionName, parameters);
+  } catch (e) {
```
Comment (herbig) [inline src/utils/contract.ts:17]: could `logError` here so we get it in Sentry if something is weird

File: src/components/ProposalCreate/TransactionsForm.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+          w="fit-content"
+          pl={0}
+        >
+          {t('labelAddTransaction')}
+        </Button>
+        <Alert
+          status="info"
+          w="fit-full"
+        >
+          <Info boxSize="24px" />
```
Comment (herbig) [inline src/components/ProposalCreate/TransactionsForm.tsx:53]: is px ok here? we generally use rem

File: src/components/ProposalCreate/Transactions.tsx [hunk 4]
```diff
@@ -8,93 +8,108 @@ import {
...
+                    p={0}
+                    textStyle="text-button-md-semibold"
+                    color="grayscale.100"
+                  >
+                    {isExpanded ? <ArrowDown boxSize="1.5rem" /> : <ArrowRight fontSize="1.5rem" />}
+                    {t('transaction')} {index + 1}
+                  </AccordionButton>
+                  {index !== 0 || transactions.length !== 1 ? (
+                    <IconButton
+                      icon={<Trash boxSize="24px" />}
```
Comment (herbig) [inline src/components/ProposalCreate/Transactions.tsx:84]: also here, rem sizing is probably better

Comment (herbig) [reply src/components/ProposalCreate/Transactions.tsx:84]: oh @Da-Colon I realized this isn't new, so probably not originally your code ha

File: src/components/ProposalCreate/TransactionsForm.tsx (hunk 3 above)
Comment (Da-Colon) [reply src/components/ProposalCreate/TransactionsForm.tsx]: yeah I like the 'rem' too. 24px === 1.5rem === 1.5 (chakra provided).

**Review by herbig — CHANGES_REQUESTED** [review]
Code looks awesome, it's great to have validation in here and it's super clean

Two issues in testing:

- [x] all 3 function fields get marked as invalid if only one of them is, which is confusing (see screenshot)
- [ ] if you input a target address, then there is a flickering on the first character input of the function name, not sure what's up with that.  I can record a video if you need

Screenshot 2023-03-17 at 10 17 35 PM

**Review by herbig — APPROVED** [review]

**Pull request merged**